www/include/Administration/myAccount/DB-Func.php

```diff
@@ -126,66 +126,68 @@ function updateContact($contact_id = null)
     $ret['contact_name'] = CentreonUtils::escapeSecure($ret['contact_name'], CentreonUtils::ESCAPE_ILLEGAL_CHARS);
     $ret['contact_alias'] = CentreonUtils::escapeSecure($ret['contact_alias'], CentreonUtils::ESCAPE_ILLEGAL_CHARS);
 
-    $rq = "UPDATE contact SET ";
-    $rq .= "contact_name = ";
-    isset($ret["contact_name"]) && $ret["contact_name"] != null
-        ? $rq .= "'" . htmlentities($ret["contact_name"], ENT_QUOTES, "UTF-8") . "', "
-        : $rq .= "NULL, ";
-    $rq .= "contact_alias = ";
-    isset($ret["contact_alias"]) && $ret["contact_alias"] != null
-        ? $rq .= "'" . htmlentities($ret["contact_alias"], ENT_QUOTES, "UTF-8") . "', "
-        : $rq .= "NULL, ";
-
-    if (isset($ret["contact_passwd"]) && $ret["contact_passwd"]) {
+    $rq = 'UPDATE contact SET ' .
+          'contact_name = :contactName, ' .
+          'contact_alias = :contactAlias, ' .
+          'contact_location = :contactLocation, ' .
+          'contact_lang = :contactLang, ' .
+          'contact_email = :contactEmail, ' .
+          'contact_pager = :contactPager, ' .
+          'default_page = :defaultPage, ' .
+          'contact_js_effects = :contactJsEffects, ' .
+          'contact_autologin_key = :contactAutologinKey';
+    $password_encrypted = null;
+    if (isset($ret['contact_passwd']) && $ret['contact_passwd']) {
+        $rq .= ', contact_passwd = :contactPasswd';
         if ($encryptType == 1) {
-            $rq .= "contact_passwd = '" . md5($ret["contact_passwd"]) . "', ";
+            $password_encrypted = md5($ret['contact_passwd']);
         } elseif ($encryptType == 2) {
-            $rq .= "contact_passwd = '" . sha1($ret["contact_passwd"]) . "', ";
+            $password_encrypted = sha1($ret['contact_passwd']);
         } else {
-            $rq .= "contact_passwd = '" . md5($ret["contact_passwd"]) . "', ";
+            $password_encrypted = md5($ret['contact_passwd']);
         }
```

Comment:
I think this part could be simplified:

if ($encryptType == 2) {
    $password_encrypted = sha1($ret['contact_passwd']);
} else {
    $password_encrypted = md5($ret['contact_passwd']);
}